### PR TITLE
Single-process and libfuzzer build options

### PR DESCRIPTION
**Pull request by fwsGonzo**
Adds varnishd support for instrumented fuzzing, PGO, callgraphs etc. by adding a single-process mode as well as an extra LIBFUZZER option. Note that I have a CMake build system I use for both varnish and varnish-plus so I didn't make any changes to autocrap.

Kinda hacky, so feel free to commit to this branch.

### CODE REVIEW

**Comment by bsdphk** [comment]
I'm not terribly keen on the way this is done.

I would expect that if you replaced fork(2) in mgt_child.c with pthread_create(3) and neutered the fd-frobbing code in the pid==0 case, things would just work ?

File: bin/varnishd/hpack/vhp_gen_hufdec.c [hunk 1]
```diff
@@ -250,5 +260,6 @@ main(int argc, const char **argv)
 	tbl_print(top);
 	printf("};\n");
 
+	tbl_free(top);
```
Comment (Dridi) [inline bin/varnishd/hpack/vhp_gen_hufdec.c]: This change should be submitted as a separate patch.

**Comment by Dridi** [comment]
This sounds more like a childless mode, we still have to fork VCC processes and fork+exec CC processes, unless I missed something in the diff?

**Comment by Dridi** [comment]
During the bugwash I suggested that childless mode and libfuzzer integration should be separate changes, I was told that childless mode was only meant for fuzzing, and then I couldn't continue the discussion.

When I see an `#ifdef SINGLE_PROCESS_MODE` block immediately followed by an `#ifndef LIBFUZZER_ENABLED` block then my intuition tells me that both could be introduced in separate commits. And as I commented earlier, you can either submit the change in `vhp_gen_hufdec.c` in a separate pull request or push it directly to master, it's self-contained and can be taken out of context.
